test/src/test/java/org/corfudb/integration/ServerRestartIT.java

```diff
@@ -810,4 +812,64 @@ public void testCorfuTableMultiIndexReconstruction() throws Exception {
         runtime2.shutdown();
         runtime3.shutdown();
     }
+
+    /**
+     * Check streamAddressSpace rebuilt from log unit
+     * when stream has been previously checkpointed and trimmed.
+     **/
+    @Test
+    public void checkStreamAddressSpaceRebuiltWithTrim() throws Exception {
```

Comment:
This test could pass in current master branch. When we introduce a new testcase for a bug fix, we should come up with a testcase that fails with current implementation, but could succeed with the fix, otherwise we cannot be sure if the fix really works, right?

Reply:
Yes, I had written a test on top of my PR 1969 and thought this would replicate it, but we need entries to be placed in between the snapshot time and the actual occurrence of the checkpoint entries. So, I've rebased on top of 1969 to have the API for option. Still I'll leave this test to catch any regression.